src/routes/LPOptimisers/useLPVaults.ts

```diff
@@ -1,53 +1,25 @@
-import { providers } from 'ethers';
 import { useEffect } from 'react';
 
 import {
-  initialiseVaultsForSignerThunk,
-  initialiseVaultsThunk,
-  resetVaultsAction,
-} from '../../app/features/lp-optimisers';
+  initialiseOptimisersThunk,
+  selectOptimisers,
+} from '../../app/features/stateless-optimisers';
 import { useAppDispatch, useAppSelector } from '../../app/hooks';
+import { useWallet } from '../../hooks/useWallet';
 
-export const useLPVaults = (signer: providers.JsonRpcSigner | null) => {
-  const vaultsLoaded = useAppSelector((state) => state.lpOptimisers.vaultsLoadedState);
-  const signerLoadedState = useAppSelector((state) => state.lpOptimisers.signerLoadedState);
-  const shouldInitVaults = vaultsLoaded === 'idle';
-  const vaultsInitialised = vaultsLoaded === 'succeeded';
-  const vaultsInitialisedWithSigner = signerLoadedState === 'succeeded';
-  const lpVaults = useAppSelector((state) => state.lpOptimisers.lpVaults);
-  const dispatch = useAppDispatch();
+export const useLPVaults = () => {
+  const { signer } = useWallet();
 
-  useEffect(() => {
-    if (!shouldInitVaults) {
-      return;
-    }
-    void dispatch(initialiseVaultsThunk());
-  }, [shouldInitVaults]);
+  const vaultsLoaded = useAppSelector((state) => state.statelessOptimisers.optimisersLoadedState);
```

Comment:
can we create a selector for this? same as selectOptimisers

Reply:
8584ee129395fff176ac73fc3ca7ff3e2c830a4b